### PR TITLE
Use ternary expression

### PR DESCRIPTION
**Pull request by RoiEXLab**
Follow-up to #5289

## Functional Changes

[ ] New map or map update
[ ] New Feature
[ ] Feature update or enhancement
[ ] Feature Removal
[x] Code Cleanup or refactor
[ ] Configuration Change
[ ] Bug fix:  
[ ] Other:   

## Testing

[x] No manual testing done
[ ] Manually tested

### CODE REVIEW

File: game-core/src/main/java/games/strategy/engine/chat/PlayerChatRenderer.java [hunk 1]
```diff
@@ -99,16 +99,12 @@ private void setIconMap() {
                 .collect(Collectors.toList());
         maxIconCounter = Math.max(maxIconCounter, icons.size());
         playerMap.put(playerNode.getPlayerName().getValue(), players);
-        if (uiContext == null) {
-          iconMap.put(playerNode.getPlayerName().getValue(), null);
-        } else {
-          iconMap.put(playerNode.getPlayerName().getValue(), icons);
-        }
+        iconMap.put(playerNode.getPlayerName().getValue(), uiContext == null ? null : icons);
```
Comment (DanVanAtta) [inline game-core/src/main/java/games/strategy/engine/chat/PlayerChatRenderer.java]: One odd thing, if `uiContext` and icons have a relationship here that is not direct or necessarily clear. EG: what happens to `icons` when uiContext is null? It seems like there is a potential for computations and then it's all ignored later when some *other* variable is null.

Would it be yet cleaner if `icons` were assigned to `null` when `uiContext` is `null`?

Comment (RoiEXLab) [reply game-core/src/main/java/games/strategy/engine/chat/PlayerChatRenderer.java]: Good point, will simplify something else even 👍

**Comment by RoiEXLab** [comment]
Ok @DanVanAtta I altered the code a bit to avoid filling the map with null values.
Instead the entries are now just omitted if the icon factory is null.

While writing this I realized that the factory can never be null in the existing code, so I will just turn this into a precondition

**Comment by RoiEXLab** [comment]
Ok, I restructured the code a little bit more.
In theory the for loop could be converted into a stream expression, but given that this would mean 3 different streams, we'd end up with more code so I decided not to do it.

File: game-core/src/main/java/games/strategy/engine/chat/PlayerChatRenderer.java [hunk 2]
```diff
@@ -74,41 +89,16 @@ private String getNodeLabelWithPlayers(final PlayerName playerName) {
...
-          iconMap.put(playerNode.getPlayerName().getValue(), null);
-        } else {
-          iconMap.put(playerNode.getPlayerName().getValue(), icons);
-        }
-      }
-    }
   }
 
-  public int getMaxIconCounter() {
-    return maxIconCounter;
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/chat/PlayerChatRenderer.java]: I double checked this is only called once per game instance, so this change is ok performance wise

**Comment by DanVanAtta** [comment]
Certainly simpler, good work @RoiEXLab  :+1:

**Pull request merged**